Applications/StableDiffusionExample/ContentView.swift

```diff
@@ -0,0 +1,310 @@
+                            Toggle("Show Progress", isOn: $showProgress)
+                        }
+                    }
+                }
+            }
+            .frame(minWidth: 300)
+        }
+        .padding()
+    }
+
+    private func generate() {
+        Task {
+            await evaluator.generate(
+                prompt: prompt, negativePrompt: negativePrompt, showProgress: showProgress)
+        }
+    }
+}
+
+/// Progress reporting with a title.
+struct Progress: Equatable {
+    let title: String
+    let current: Double
+    let limit: Double
+}
+
+/// Async model factory
+actor ModelFactory {
+
+    enum LoadState {
+        case idle
+        case loading(Task<ModelContainer<TextToImageGenerator>, Error>)
+        case loaded(ModelContainer<TextToImageGenerator>)
+    }
+
+    enum SDError: Error {
+        case unableToLoad
+    }
+
+    public nonisolated let configuration = StableDiffusionConfiguration.presetSDXLTurbo
+
+    /// if true we show UI that lets users see the intermediate steps
+    public nonisolated let canShowProgress: Bool
+
+    /// if true we show UI to give negative text
+    public nonisolated let canUseNegativeText: Bool
+
+    private var loadState = LoadState.idle
+    private var loadConfiguration = LoadConfiguration(float16: true, quantize: false)
+
+    public nonisolated let conserveMemory: Bool
+
+    init() {
+        let defaultParameters = configuration.defaultParameters()
+        self.canShowProgress = defaultParameters.steps > 4
+        self.canUseNegativeText = defaultParameters.cfgWeight > 1
+
+        // this will be true e.g. if the computer has 8G of memory or less
+        self.conserveMemory = MLX.GPU.memoryLimit < 8 * 1024 * 1024 * 1024
+
+        if conserveMemory {
+            print("conserving memory")
+            loadConfiguration.quantize = true
+            MLX.GPU.set(cacheLimit: 1 * 1024 * 1024)
+            MLX.GPU.set(memoryLimit: 3 * 1024 * 1024 * 1024)
+        } else {
+            MLX.GPU.set(cacheLimit: 256 * 1024 * 1024)
+        }
+    }
+
+    public func load(reportProgress: @escaping @Sendable (Progress) -> Void) async throws
+        -> ModelContainer<TextToImageGenerator>
+    {
+        switch loadState {
+        case .idle:
+            let task = Task {
+                try await configuration.download { progress in
+                    if progress.fractionCompleted < 0.99 {
+                        reportProgress(
+                            .init(
+                                title: "Download", current: progress.fractionCompleted * 100,
+                                limit: 100))
+                    }
+                }
+
+                let container = try ModelContainer<TextToImageGenerator>.createTextToImageGenerator(
+                    configuration: configuration, loadConfiguration: loadConfiguration)
+
+                await container.setConserveMemory(conserveMemory)
```

Comment:
The container is the actor (thread safe) mechanism that holds the loaded model.  Signal if we want to conserve memory -- this will cause the container to discard the first half of the model when preparing the image itself.